add InvalidCredentialsError with language-agnostic code

Hafizzle opened the pull request:
add InvalidCredentialsError with language-agnostic code and update backend & web logout logic

Hafizzle committed:
add InvalidCredentialsError with language-agnostic code and update backend & web logout logic

Hafizzle committed:
error.code in API error responses

Updated the error handler to serialize responseErr.code as the JSON error field, falling back to responseErr.name when no code is set.

SabreCat committed:
fix(lint): whitespace and missing def

SabreCat committed:
fix(lint): missed one

SabreCat commented:
Looks good after lint corrections, though this would be a good opportunity to add some test coverage! Even one case covering what an invalid credentials response from the auth middleware ought to look like, would be an improvement.

Hafizzle committed:
add InvalidCredentialsError case for bad token

Add test verifying that auth middleware throws InvalidCredentialsError with code "invalid_credentials" and correct translated message when the API token is invalid.

SabreCat committed:
fix(test): user fields implicitly required

phillipthelen reviewed (approved):
Looks good to me!
